addons/mail/security/mail_security.xml

```diff
@@ -276,4 +276,29 @@
             <field name="perm_create" eval="True"/>
             <field name="perm_unlink" eval="True"/>
         </record>
+
+        <record id="ir_rule_mail_canned_response_admin_rule" model="ir.rule">
+            <field name="name">Canned response: admin access</field>
+            <field name="model_id" ref="model_mail_canned_response"/>
+            <field name="groups" eval="[Command.link(ref('group_mail_canned_response_admin'))]"/>
+            <field name="domain_force">[(1, '=', 1)]</field>
+        </record>
+
+        <record id="ir_rule_mail_canned_response_creator_rule" model="ir.rule">
+            <field name="name">Canned response: User access</field>
+            <field name="model_id" ref="model_mail_canned_response"/>
+            <field name="groups" eval="[Command.link(ref('base.group_user'))]"/>
+            <field name="domain_force">[('create_uid', '=', user.id)]</field>
```

Comment:
I understand nothing of the above. The permission setup for this PR is a nightmare and since it looks like it's working as expected (I still need feedback on this)  I don't wish to touch anything.

Reply:
I m sorry, but all of those rule look wrong

If it's your canned response (`'create_uid', '=', user.id`), you should be able to do what ever you want. In your code, it look like you can't read them. 

I feel like you really like complex solution to simple problem, if you look at my rules, you could break them like 

- Admin group => no restriction
- canned response author => no restriction
- shared canned response => only read permission

I would argue that explicitly making `perm_*` to True would improve the code, but it's more LoC so I didn't.

Reply:
Please explain what prevent from reading your own canned response in the unified rule.

Reply:
Sur thing:

```xml
        <record id="ir_rule_mail_canned_response_user_update" model="ir.rule">
            <field name="name">Canned response: User write/unlink: own only</field>
            <field name="model_id" ref="model_mail_canned_response"/>
            <field name="groups" eval="[Command.link(ref('base.group_user'))]"/>
            <field name="domain_force">[('create_uid', '=', user.id)]</field>
            <field name="perm_create" eval="False"/>
            <field name="perm_read" eval="False"/>
        </record>
```

If this rule doesn't mean "the creator of the canned response cannot read them" then I don't know what I'm looking at.

Reply:
It means "this rule does not apply to read permission check". This rule therefore applies to write and unlink. "perm_something" is a boolean field telling in which case the rule is applied.

So please next time I suggest you something, don't send me to hell. Maybe you don't like my reviews but those are based on some experience. I am open to discussion, I make mistakes like everyone, but consider I don't ask for changes just to annoy you.

:kissing_heart:

Reply:
> It means "this rule does not apply to read permission check". This rule therefore applies to write and unlink. "perm_something" is a boolean field telling in which case the rule is applied.

Then I'm wrong to assume that something named `perm_*` is a permission, because it's a filter to determine how to apply the rule.
Do you see why I'm confused?
Because to me, this rule reads like this: if the `create_uid == user.id` and we try to read the canned response => reject.

I'm not sending you to hell, you are making suggestions that don't make sense to me, based on the knowledge I have.